src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -350,13 +350,19 @@ public ResponseEntity<byte[]> getLecturePdfAttachmentsMerged(@PathVariable Long
             return ResponseEntity.status(HttpStatus.FORBIDDEN).body(errorMessage.getBytes());
         }
 
+        // Parse user information from the access token
+        var claims = this.tokenProvider.parseClaims(temporaryAccessToken);
+        String username = claims.getSubject();
+        User user = userRepository.getUserWithGroupsAndAuthorities(username);
+
         Set<AttachmentUnit> lectureAttachments = attachmentUnitRepository.findAllByLectureIdAndAttachmentTypeElseThrow(lectureId, AttachmentType.FILE);
 
         List<String> attachmentLinks = lectureAttachments.stream()
-                .filter(unit -> unit.isVisibleToStudents() && "pdf".equals(StringUtils.substringAfterLast(unit.getAttachment().getLink(), ".")))
+                .filter(unit -> authCheckService.isAllowedToSeeLectureUnit(unit, user) && "pdf".equals(StringUtils.substringAfterLast(unit.getAttachment().getLink(), ".")))
```

Comment:
If you pass `null` as the user, the method itself will fetch it from the `userRepository`. Sure, you have some overhead from the loop -- on the other side you don't need to inject the repository into this class. What performs better?